src/PeakFinder.cpp

```diff
@@ -0,0 +1,170 @@
+            uint64_t yRight = y[xRight];
+
+            // Right peak is bigger
+            if (yRight > yLeft){
+                rightPeak.left = leftPeak.left;
+                peakIndex[i] = peakIndex[i + 1];
+
+                // Update the bounds of the dead peak so it stops (inclusive) at the merge point
+                leftPeak.right = i;
+
+                // Update boundary peak index
+                peakIndex[leftPeak.left] = peakIndex[i + 1];
+                peakIndex[leftPeak.right] = peakIndex[i + 1];
+
+                // Update the weaker peak to reflect that is has terminated
+                leftPeak.stop = i;
+                leftPeak.isMerged = true;
+                leftPeak.persistence = y[rightPeak.start] - y[i];
+            }
+            // Left peak is bigger, or same size
+            else{
+                leftPeak.right = rightPeak.right;
+                peakIndex[i] = peakIndex[i - 1];
+
+                // Update the bounds of the dead peak so it stops (inclusive) at the merge point
+                rightPeak.left = i;
+
+                // Update boundary peak index
+                peakIndex[rightPeak.right] = peakIndex[i - 1];
+                peakIndex[rightPeak.left] = peakIndex[i - 1];
+
+                // Update the weaker peak to reflect that is has terminated
+                rightPeak.stop = i;
+                rightPeak.isMerged = true;
+                rightPeak.persistence = y[rightPeak.start] - y[i];
+            }
+        }
+    }
+
+    // Set the persistence for the tallest peak as its total height from 0
+    peaks[0].persistence = y[peaks[0].start];
+}
+
+
+void PeakFinder::sortByPersistence(){
+    // Sort the peaks based on their persistence
+    sort(peaks.begin(), peaks.end(),
+         [&](const Peak& a, const Peak& b) {
+             // Equal values are sorted by lowest x value
+             if (a.persistence == b.persistence){
+                 return (a.start < b.start);
+             }
+             else {
+                 return (a.persistence > b.persistence);
+             }
+         }
+    );
+}
+
+
+uint64_t PeakFinder::calculateArea(vector<uint64_t>& y, uint64_t x_min, uint64_t x_max){
+    uint64_t total = 0;
+
+    for (size_t i=x_min; i<=x_max; i++){
+        total += y[i];
+    }
+
+    return total;
+}
+
+
+uint64_t PeakFinder::findXCutoff(vector<uint64_t>& y){
+    sortByPersistence();
+
+    // find the total AUC
+    uint64_t total_area = calculateArea(y, 1, y.size() - 1);
+
+    // Second most prominent peak
+    Peak& p = peaks[1];
+
+    // Find the AUC inside the bounds of the peak (from y=0 and up)
+    uint64_t peakArea = calculateArea(y, p.left, p.right);
+
+    double percentArea = 100 * (double(peakArea) / double(total_area));
+
+    uint64_t x_cutoff;
+
+    // Check if second most prominent peak is reasonable size (not a false peak)
+    if (percentArea > 1){   //TODO: readjust this proportion for histograms with 0=0 frequency
```

Comment:
Remove the comment if this is no longer needed. Or, if necessary, expand the comment to a multiple line comment.